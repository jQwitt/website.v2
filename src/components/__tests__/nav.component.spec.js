import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { Navigation } from '../index';

afterEach(cleanup);
describe('Navigation renders correctly', () => {
  test('snapshot', () => {
    const { asFragment } = render(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('toggles the nav bar', () => {
    const { container, getByTestId } = render(<Navigation />);
    fireEvent.click(getByTestId('togglesIcon'));
    expect(
      getByTestId('containerPageLinks').classList.contains('hidden-mobile')
    ).toBeFalsy();
    expect(
      getByTestId('containerSocialLinks').classList.contains('hidden-mobile')
    ).toBeFalsy();
    fireEvent.click(getByTestId('togglesIcon'));
    expect(
      getByTestId('containerPageLinks').classList.contains('hidden-mobile')
    ).toBeTruthy();
    expect(
      getByTestId('containerSocialLinks').classList.contains('hidden-mobile')
    ).toBeTruthy();
  });

  it('renders correct items', () => {
    const { getByText } = render(<Navigation />);
    expect(getByText('website.v2')).toBeTruthy();
    expect(getByText('experience')).toBeTruthy();
    expect(getByText('projects')).toBeTruthy();
    expect(getByText('education')).toBeTruthy();
    expect(getByText('contact')).toBeTruthy();
  });
});
