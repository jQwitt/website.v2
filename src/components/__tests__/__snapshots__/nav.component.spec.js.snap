// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-xl-3 col-lg-4 col-12 py-lg-3 row navigation position-fixed"
  >
    <div
      class="col-md-12 align-self-start"
    >
      <h1
        class="underlined padded"
      >
        website.v2
      </h1>
      <div
        class="navigation-links hidden-mobile"
        data-testid="containerPageLinks"
      >
        <div>
          <h3>
            experience
          </h3>
        </div>
        <div>
          <h3>
            projects
          </h3>
        </div>
        <div>
          <h3>
            education
          </h3>
        </div>
        <div>
          <h3>
            contact
          </h3>
        </div>
      </div>
    </div>
    <div
      class="col-12 align-self-end bottom-right-mobile hidden-mobile"
      data-testid="containerSocialLinks"
    >
      <div
        class="social-media-links d-flex justify-content-center"
      >
        <a
          href="https://www.linkedin.com/in/joseph-witt-52a960171/"
          name="linkedin"
        >
          <i
            class="flaticon-linkedin medium"
          />
        </a>
        <a
          href="mailto:jackwittbiz@gmail.com"
          name="email"
        >
          <i
            class="flaticon-email medium"
          />
        </a>
        <a
          href="https://github.com/jQwitt"
          name="github"
        >
          <i
            class="flaticon-browser medium"
          />
        </a>
      </div>
    </div>
    <div
      class="top-right-mobile hidden-desktop"
      data-testid="togglesContainer"
    >
      <div
        class=" icon"
      >
        <i
          class="flaticon-add medium"
          data-testid="togglesIcon"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
